Samples/Tutorial Samples/WinMLPIXSample/WinMLPIXSample/main.cpp

```diff
@@ -0,0 +1,129 @@
+            break;
+        }
+
+        // Check to see if the adapter supports Direct3D 12, but don't create the
+        // actual device yet.
+        if (SUCCEEDED(D3D12CreateDevice(pAdapter, D3D_FEATURE_LEVEL_12_0, _uuidof(ID3D12Device), nullptr)))
+        {
+            *ppAdapter = pAdapter;
+            return;
+        }
+        pAdapter->Release();
+    }
+}
+
+PIXCaptureParameters GetCap()
+{
+    PIXCaptureParameters cap;
+    cap.GpuCaptureParameters.FileName = pixCaptureFilePath.c_str();
+    return cap;
+}
+
+// Create command queue & command list to be used by WinML
+void GetD3D12CommandAssets()
+{
+    D3D12_COMMAND_LIST_TYPE type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+    D3D12_COMMAND_QUEUE_DESC queueDesc = {};
+    queueDesc.Type = type;
+    queueDesc.Flags = D3D12_COMMAND_QUEUE_FLAG_NONE;
+    ID3D12CommandAllocator* commandAllocator;
+
+    IDXGIFactory4* factory;
+    CreateDXGIFactory1(IID_PPV_ARGS(&factory));
+
+    IDXGIAdapter1* hardwareAdapter;
+    GetHardwareAdapter(factory, &hardwareAdapter);
+
+    D3D12CreateDevice(hardwareAdapter, D3D_FEATURE_LEVEL_12_0, _uuidof(ID3D12Device), (void**)&m_d3d12Device);
+
+    m_d3d12Device->CreateCommandAllocator(
+        type,
+        IID_PPV_ARGS(&commandAllocator)
+    );
+    m_d3d12Device->CreateCommandQueue(&queueDesc, IID_PPV_ARGS(&m_commandQueue));
+    m_d3d12Device->CreateCommandList(0, type, commandAllocator, nullptr, __uuidof(ID3D12CommandList), (void**)&m_commandList);
+}
+
+void LoadAndEvaluate(ID3D12CommandQueue* commandQueue)
+{
+    // Setting markers for each step, these markers will split the commands into sections for easier debugging
+    PIXSetMarker(commandQueue, m_color, "Start loading model...");
+    LoadModel();
+
+    PIXSetMarker(commandQueue, m_color, "Start loading image...");
+    LoadImageFile(imagePath);
+
+    PIXSetMarker(commandQueue, m_color, "Start creating session...");
+    CreateSession(commandQueue);
+
+    PIXSetMarker(commandQueue, m_color, "Start binding model...");
+    BindModel();
+
+    PIXSetMarker(commandQueue, m_color, "Start evaluating model...");
+    EvaluateModel();
+
+}
+
+void CaptureWithUserSetMarker()
+{
+    // Create command queue and command list, this command queue will be used to create model 
+    // device and then passed into PIXBeginEvent, PIXEndEvent and PIXSetMarker methods to set 
+    // markers. 
+    GetD3D12CommandAssets();
+
+    PIXCaptureParameters capParams = GetCap();
+
+    // Start the GPU capture
+    PIXBeginCaptureRedist(PIX_CAPTURE_GPU, &capParams);
+
+    // Start PIX event, markers can only be set within Being and End event sections
+    PIXBeginEvent(m_commandQueue, m_color, "WinMLPIXSample");
+
+    // Do the ML computation
+    LoadAndEvaluate(m_commandQueue);
+
+    // End PIX event
+    PIXEndEvent(m_commandQueue);
+
+    // End capture
+    PIXEndCaptureRedist(FALSE);
+}
+
+int main()
+{
+    TryEnsurePIXFunctions();
+    init_apartment();
+    CaptureWithUserSetMarker();
```

Comment:
>CaptureWithUserSetMarker [](http://example.com/codeflow?start=4&length=24)

How do you see the results of this capture? Do you still need to attach PIX to the executable?

Reply:
Ah, yes, I should mention in the comments that there will be a .wpix file generated when the program is run.

Reply:
Also added a message in the console output.